### ISSUE TITLE
accommodate bazel changes

### ISSUE THREAD

**Issue by sreev** [apache/incubator-heron#3113]
https://blog.bazel.build/2018/08/22/bazel-homebrew.html

- upgrade bazel, per instructions on above link
- change heron build instructions

**Comment by sreev** [comment]
```
$ brew uninstall --force bazel
Uninstalling bazel... (48 files, 402.5MB)

$ brew tap bazelbuild/tap
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 2 taps (homebrew/core and homebrew/cask).
==> Tapping bazelbuild/tap
Cloning into '/usr/local/Homebrew/Library/Taps/bazelbuild/homebrew-tap'...
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (9/9), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 9 (delta 0), reused 4 (delta 0), pack-reused 0
Unpacking objects: 100% (9/9), done.
Tapped 2 formulae (38 files, 47.0KB).

$ brew tap-pin bazelbuild/tap
==> Pinned bazelbuild/tap

$ brew install bazel
To restore the stashed changes to /usr/local/Homebrew run:
  'cd /usr/local/Homebrew && git stash pop'
==> Downloading https://homebrew.bintray.com/bottles/bazel-0.18.1.mojave.bottle.tar.gz
######################################################################## 100.0%
==> Pouring bazel-0.18.1.mojave.bottle.tar.gz
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d

zsh completions have been installed to:
  /usr/local/share/zsh/site-functions
==> Summary
🍺  /usr/local/Cellar/bazel/0.18.1: 12 files, 117.4MB

$ bazel version
WARNING: Processed legacy workspace file incubator-heron/tools/bazel.rc. This file will not be processed in the next release of Bazel. Please read https://github.com/bazelbuild/bazel/issues/6319 for further information, including how to upgrade.
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
Build label: 0.18.1-homebrew
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Fri Nov 2 11:16:42 2018 (1541157402)
Build timestamp: 1541157402
Build timestamp as int: 1541157402

```

**Comment by sreev** [comment]
build successful on macOS Mojave Version 10.14, java version "1.8.0_181".

**Comment by sreev** [comment]
```
added new file %WORKSPACE%/.bazelrc
with one line:
import tools/bazel.rc

$ bazel build --config=darwin heron/...
WARNING: Duplicate rc file: /Users/svaddi/SreeVaddi/source/github/sreev/incubator-heron/tools/bazel.rc is read multiple times, most recently imported from /Users/svaddi/SreeVaddi/source/github/sreev/incubator-heron/.bazelrc
WARNING: Processed legacy workspace file /Users/svaddi/SreeVaddi/source/github/sreev/incubator-heron/tools/bazel.rc. This file will not be processed in the next release of Bazel. Please read https://github.com/bazelbuild/bazel/issues/6319 for further information, including how to upgrade.
Starting local Bazel server and connecting to it...
...
...
...
INFO: From Executing extra_action //tools/python:checkstyle_python on //heron/tools/tracker/tests/python:query_operator_unittest_runner:
Nov 19, 2018 12:27:50 AM org.apache.bazel.checkstyle.PythonCheckstyle main
INFO: 85 python files found by checkstyle
INFO: Elapsed time: 946.638s, Critical Path: 179.46s
INFO: 2252 processes: 1921 local, 331 worker.
INFO: Build completed successfully, 3926 total actions

```

**Comment by rohanag12** [comment]
Fixed by #3209

**Comment by Code0x58** [comment]
looks like another one to close off
